DS-1.1 DQC and ETL

ArvinDodangeh opened the pull request.

ArvinDodangeh committed:
fix picture bug in JN

natallia-zzz commented:
Please change this to a real pull request, not a draft.

natallia-zzz commented:
Its still a draft. I cannot review a draft

There are 2 notebooks. What do I look at? 

Also your repository must be structured and not just a pile of documents. Think of how you want to structure it for convenient use. 

for example: 
project/
&nbsp; data/
 &nbsp; notebooks/
 &nbsp; src /

ArvinDodangeh committed:
Restructure repository with data, notebooks, and src folders

ArvinDodangeh commented:
> Its still a draft. I cannot review a draft
> 
> There are 2 notebooks. What do I look at?
> 
> Also your repository must be structured and not just a pile of documents. Think of how you want to structure it for convenient use.
> 
> for example: project/   data/   notebooks/   src /


I fixed draft and project structure .

ArvinDodangeh committed:
Delete notebooks/DQC_initial.ipynb

natallia-zzz commented:
Good: structuring notebook and adding conclusions. 

General remarks: 
some outputs such as missing values are hard to read. For example df validity. What is the output there? and also missing values graphs are uninterpretable. How can you tell that there are no missing values? they do not give 

DQC issues: 
- only looked into numerical features. What about categorical and time features? 
- What is the checking consistency 2 for? 
- Very general checks. Remember that these are sales data, the data must match the sales logic. For example there is a negative price value and item count? This was not checked at all

** There is no ETL **

ArvinDodangeh committed:
Improve Data Quality Checks:


- Added checks for categorical and time features.
- Enhanced consistency checks for sales logic.
- Validated negative values in price and item count.

ArvinDodangeh committed:
Added ETL to nb

ArvinDodangeh committed:
Rename DQC_v3_[Include ETL] .ipynb to dqc-and-etl(1) .ipynb

Rename to original file name !

ArvinDodangeh committed:
Adding ETL script

ArvinDodangeh committed:
Final Version of DQC

ArvinDodangeh committed:
Delete unnecessary

Delete unnecessary notebooks

ArvinDodangeh committed:
Merge branch 'DS-1.1' of https://github.com/ArvinDodangeh/DS into DS-1.1

ArvinDodangeh committed:
Add data to   directory

natallia-zzz commented:
Fix: 
- comments should be in 3rd person. "We will", "as you can see", "I think", etc. is does not meet the requirements for report style of writing. If needed, read about the different styles of speech: small talk, formal talk, report style, scientific style, how to write emails and etc. 
- How can we tell if negative values are returns or errors in the database? And why is it recommended to delete them? 
- Date consistency check is uninterpretable: the histogram does not confirm that there all of the intervals between days is 1 day. Most of them yes, but what if there are a couple of missed days? It cannot be confirmed by the histogram. 
- sales duplicates: the function duplicated checks for rows being repeated. So the outputed 6 rows do not mean that they are duplicates for each other. Research the duplicated() function in pandas and redo the check. Also ask the question "what to duplicates do in this specific task?" 

Other comments and suggestions for the future
- there is no need to check the format of the datetime. As long as it's interpretable/convertable and consistent throughout the dataset. This is just a comment, the code does not need to be deleted. 
- printing a dict is not a good way to present some information. It is hard to read. Think of some ways the information can be interpreted and visualized. This doesn't always mean graphs. It could be text, tables, etc.  
- The Pull request should be called with the task name "DS-1.1 DQC and ETL"

ArvinDodangeh committed:
Update ETL.py

Decision for aggregating item_cnt_day  has been postponed after EDA ( for further analysis)

ArvinDodangeh committed:
Kaggle Notebook | DQC and ETL | Version 1

natallia-zzz commented:
- Why did you assume that missing values in test in intentional? Test was checked after, so this note should be deleted
- Date consistency-> very good
- Looking at dynamics is part of EDA. This should not be part of EDA unless it proves a point(such as negative values are returns) 
- Are the outliers reasonable? (item price and item counts)

ArvinDodangeh committed:
Kaggle Notebook | DQC and ETL | Version 1

natallia-zzz commented:
- How does HDBSCAN tell that these are errors? Is it a valid algorthm to use. 
- Are the outliers reasonable? (item price and item counts)
- Do not leave commented chunks of code in a pull request, please
- Do not use a big block of mixed text as your report output. Think of ways to make it more interpretable

ArvinDodangeh committed:
Kaggle Notebook | DQC and ETL | Version 1
